src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -397,6 +397,34 @@ def parse_shared_gallery_image_id(image_reference):
     return image_info.group(1), image_info.group(2)
 
 
+def is_community_gallery_image_id(image_reference):
+    if not image_reference:
+        return False
+
+    community_gallery_id_pattern = re.compile(r'^/CommunityGalleries/[^/]*/Images/[^/]*/Versions/.*$', re.IGNORECASE)
+    if community_gallery_id_pattern.match(image_reference):
+        return True
+
+    return False
+
+
+def parse_community_gallery_image_id(image_reference):
+    from azure.cli.core.azclierror import InvalidArgumentValueError
+
+    if not image_reference:
+        raise InvalidArgumentValueError(
+            'Please pass in the community gallery image id through the parameter --image')
+
+    image_info = re.search(r'^/CommunityGalleries/([^/]*)/Images/([^/]*)/Versions/.*$', image_reference, re.IGNORECASE)
+    if not image_info or len(image_info.groups()) < 2:
```

Comment:
Why not check image version ? 
```suggestion
    image_info = re.search(r'^/CommunityGalleries/([^/]*)/Images/([^/]*)/Versions/(.*)$', image_reference, re.IGNORECASE)
    if not image_info or len(image_info.groups()) < 3:
```

Reply:
At present, we need to parse the `public_gallery_name` and `gallery_image_name` from community gallery image id to query the community gallery image info, but version is not used for the time being, so we do not further match version.
https://github.com/Azure/azure-cli/blob/ddb1665b7baa01fe3ec7d75b846a4c89a2d832d6/src/azure-cli/azure/cli/command_modules/vm/_validators.py#L565-L570

I think we can add version matching when we need to parse the version. What do you think?